searchsummary/src/vespa/searchsummary/docsummary/attribute_field_writer.h

```diff
@@ -27,7 +27,7 @@ class AttributeFieldWriter
     virtual ~AttributeFieldWriter();
     virtual void fetch(uint32_t docId) = 0;
     virtual void print(uint32_t idx, vespalib::slime::Cursor &cursor) = 0;
-    static std::unique_ptr<AttributeFieldWriter> create(vespalib::Memory fieldName, const search::attribute::IAttributeVector &attr);
+    static std::unique_ptr<AttributeFieldWriter> create(vespalib::Memory fieldName, const search::attribute::IAttributeVector &attr, bool no_empty_strings = false);
```

Comment:
Consider renaming to _keep_empty_strings_, as I think that is easier to read. If so, also consider renaming WriteStringFieldNeverSkip to WriteStringFieldKeepEmpty or similar.